bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java

```diff
@@ -89,6 +94,14 @@ public String getThingID() {
         return this.thingID;
     }
 
+    public AndroidTVDynamicStateDescriptionProvider getStateDescriptionProvider() {
```

Comment:
Unfortunately I didn't spot this in my quick review earlier today, but this is not used either - see https://github.com/openhab/openhab-addons/pull/16191#discussion_r1448719145 - and should be removed as well.

 I now understand what you mean by "Prepares for PhilipsTV", but this is not the right way. You mentioned also Sony Scalar as well.

If you had only the PhilipsTV PR pending depending on this change, I would suggest to simply keep the change in that PR as you did originally. In case you will also need exactly the same ground work in your other upcoming PR, you could do the changes in a local branch and branch out to both branches from that, so that each branch would include it, but at the same time make it easier to manage separately (i.e. you could make changes in the original branch and rebase the other two against that). The first PR that gets merged, will introduce it, and the other one can be rebased then.

So in other words, I would keep this PR self-contained, i.e. probably strictly as a bugfix PR and not introduce any methods that are not currently used. It's very hard to review/assess the need without looking at the others PR's as well, and if they would never happen (for any reason) this code makes no sense.

Reply:
I see your point, I'll pull those out.  My problem with Philips and soon to be Sony is that they make changes to exactly the same lines of code so it's hard to maintain the two branches.  For example, in the thing factory I would be adding a thing type for both items.  They will end up on the same line in there.  Is it then your suggestion to create effectively 3 PRs?  One for Philips, one for Sony, and one with the "both need this" like the thing types?

Reply:
My suggestion is to create only two PR's, but your two corresponding branches could be branched out from a common branch introducing this groundwork, i.e.:
- main
  - androidtv-preparations
    - androidtv-philips
    - androidtv-sony

When you create the PR's, each will drag the preparations along with them, but they will not depend on each other, so they won't have to merged in a particular order. When one of them is merged, you can rebase and resolve the few conflicts where lines are changed just next to each other. While you are working on both, you can still make changes directly to `androidtv-preparations` and rebase the other two against that to get the latest changes into each branch.

This is just some brainstorming on how I could imagine managing this situation. You might have other solutions/preferences, and that's fine as well. The important thing would be to try to keep things separate and self-contained.

Reply:
That makes sense.  I've pushed a commit to remove as requested.  An alternative (hope) in this case would be to just get the philips reviewed and merged in since it's "more or less" done.  At that point this isn't an issue as it would all be merged prior to Sony being submitted.